src/script/conversation/ConversationRepository.ts

```diff
@@ -2068,21 +2069,13 @@ export class ConversationRepository {
       ? {domain: '', id: eventData.conversationId}
       : qualified_conversation || {domain: '', id: conversation};
 
+    if (type === CONVERSATION_EVENT.MLS_WELCOME_MESSAGE) {
+      this.handleMLSWelcomeMessageEvent(eventJson);
+    }
+
     const inSelfConversation = this.conversationState.isSelfConversation(conversationId);
     if (inSelfConversation) {
-      const typesInSelfConversation = [
-        CONVERSATION_EVENT.MEMBER_UPDATE,
-        ClientEvent.CONVERSATION.MESSAGE_HIDDEN,
-        /**
-         * As of today (07/07/2022) the backend sends `WELCOME` message to the user's own
-         * conversation (not the actual conversation that the welcome should be part of)
-         */
```

Comment:
is this not true anymore? Is it sent in the real conversation now?

Reply:
Yes, the `conversation` field on welcome message event now contains the id of the actual conversation and not the id of the self conversation as it was before.

<img width="1101" alt="Screenshot 2023-08-07 at 21 20 31" src="https://github.com/wireapp/wire-webapp/assets/45733298/7350f943-7ed9-4280-8108-6c4eaace6cea">

Reply:
great news!!! 👐

Reply:
@atomrc and then the next step is to fetch conversation if it was not known locally as usual, see https://github.com/wireapp/wire-webapp/blob/dev/src/script/conversation/ConversationRepository.ts#L2101. It was not possible previously so we had to wait for `conversation.member.join` event in case welcome arrived first. Right now it doesn't really matter which one comes first.